apps/overige-objecten-api/src/controllers/objects/index.ts

```diff
@@ -25,70 +38,94 @@ export const getAllObjectsController: RequestHandler = async (req, res, next) =>
+        const vacUrl = new URL(`/api/v2/objects/${item.attributes.vac.uuid}`, serverURL).href;
+        return {
+          uuid: item.attributes.vac.uuid,
+          type: vacSchemaURL,
+          url: vacUrl,
+          record: {
+            index: parseInt(item.id, 10),
+            startAt: item.attributes.createdAt,
+            typeVersion: 1,
+            data: {
+              ...item.attributes.vac,
+              url: vacUrl,
+            },
+            geometry: null,
+            endAt: null,
+            registrationAt: item.attributes.createdAt,
+          },
+        };
+      });
+      return vac;
+    };
     // Set response content type
     res.set('Content-Type', 'application/json');
-
     // Send results based on the requested type
-    if (type.endsWith('kennisartikel')) return res.status(200).json({ ...pagination, results: kennisartikel });
-
-    if (type.endsWith('vac')) return res.status(200).json({ results: vac });
+    if (isKennisartikel) {
+      const data = await fetchKennisartikelen();
+      const pagination = await getPaginatedResponse(req, data?.products);
+      const kennisartikelData = getKennisartikelData({ data });
+      return res.status(200).json({ ...pagination, results: kennisartikelData });
+    }
+    if (isVac) {
+      const data = await fetchVac();
+      const vac = getVacData(data);
+      const pagination = await getPaginatedResponse(req, data?.vacs as any);
+      return res.status(200).json({ ...pagination, results: vac });
+    }
+    const productsData = await fetchKennisartikelen();
+    const data = await fetchVac();
+    const kennisartikelPagination = await getPaginatedResponse(req, productsData?.products);
+    const vacPagination = await getPaginatedResponse(req, data?.vacs as any);
+    const isCount = kennisartikelPagination?.count && vacPagination?.count;
+    const isTotal = kennisartikelPagination?.total && vacPagination?.total;
+    const pagination = {
+      ...kennisartikelPagination,
+      ...vacPagination,
+      count: isCount ? (kennisartikelPagination?.count as number) + (vacPagination?.count as number) : undefined,
+      total: isTotal ? (kennisartikelPagination?.total as number) + (vacPagination?.total as number) : undefined,
+    };
+    const kennisartikelData = getKennisartikelData({ data: productsData });
+    const vac = getVacData(data);
 
-    // If no specific type, return both kennisartikel and vac objects
-    if (vac.length > 0 && kennisartikel.length > 0)
-      return res.status(200).json({ ...pagination, results: [...kennisartikel, ...vac] });
-    if (vac.length > 0) return res.status(200).json({ ...pagination, results: [...vac] });
-    if (kennisartikel.length > 0) return res.status(200).json({ ...pagination, results: [...kennisartikel] });
-    return res.status(200).json({ ...pagination, results: [] });
+    return res.status(200).json({ ...pagination, results: [...kennisartikelData, ...vac] });
```

Comment:
Het voelt voor mij nog steeds alsof er onnodig veel `res.status(200)` calls zijn. Kunne we niet zoiets doen?

```js
// begin van de functie
let results = [];
const kennisartikel = type == 'kennisartikel' || !type;
const vac = type == 'vac' || !type;

if (kennisArtikel) {
  results = [...await fetchKennisartikel())
};

if (vac) {
  results = [...await fetchVac())
};

// stuff

return res.status(200).json({ ...pagination, results });
```